internal/form/form.go

```diff
@@ -0,0 +1,121 @@
+package form
+
+import (
+	"encoding/json"
+	"net/url"
+	"reflect"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	ErrInvalidType      = errors.New("invalid type")
+	ErrUnsupportedField = errors.New("unsupported field")
+	ErrMissingRequired  = errors.New("missing required field")
+)
+
+type Message interface {
```

Comment:
hmm i don't love form.Message. Message isn't really defined in our system as "something that can be form-encoded" but is intended to be something like protobuf's Message i.e. a generic struct meant for use as param and return.

what if we move both of these under api? Message in the top-level pkg with form nested underneath?